src/autoscaling/autoscaled_pool.js

```diff
@@ -114,6 +114,19 @@ const DEFAULT_OPTIONS = {
  *   Options to be passed down to the {@link SystemStatus} constructor. This is useful for fine-tuning
  *   the system status reports. If a custom snapshotter is set in the options, it will be used
  *   by the pool.
+ *
+ * @property {Number} minConcurrency
+ *   The minimum number of tasks running in parallel.
+ *   To change this property safely, use the {@link setMinConcurrency} function.
+ * @property {Number} maxConcurrency
+ *   The maximum number of tasks running in parallel.
+ *   To change this property safely, use the {@link setMaxConcurrency} function.
+ * @property {Number} desiredConcurrency
```

Comment:
Do we need the `@property` docs? It describes private properties, will make the page more bloated and only repeats what is already said elsewhere.

Reply:
If there are set functions, there should also be a way to obtain the current value, users shouldn't be forced to store it separately.

Reply:
That's a fair point, but then I'd either add a `getSomething` function or make the properties public. It does not feel right to set them one way and get them differently. Or we can use the [Javascript getters and setters API](https://javascript.info/property-accessors)

Reply:
Good point, getters/setters would be ideal, but we should keep the `setXxx()` functions available for backwards compatibility.

Reply:
As it's not possible to have public and private properties in JS, I would consider all properties to be private and implement the interface to all public properties as setProperty() and getProperty() methods.

Reply:
It's coming really soon :) https://github.com/tc39/proposal-class-fields

Reply:
Why not use getters and setters? If the properties are called like `_desiredCapacity` and there's a getter/setter for `desiredCapacity`, everything will be alright, not?

Reply:
I don't have a preference. Native Getters/Setters look much better imo, but `setDesiredCapacity()` will fail with a meaningful error when you mistype the name like `setDesirredCapacity(5)`, but `pool.desirredCapacity = 5` will just silently create a new property on the `pool` object. Unless we `Object.freeze()` all our class instances.